Added new statistic attributes

sti0 opened the pull request:
## Description:
Added new attributes:
  - Cleaning count
  - Total cleaning time
  - Total cleaning area
  - Time left to change main brush, side brush and filter

## Checklist:

  - [X] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  ![2017-09-14 23_11_18-home assistant](https://user-images.githubusercontent.com/18382402/30455967-1b0ad1ac-99a2-11e7-81a7-dd8d4f1e4da6.png)


***This is my first pull request and github adventure, please be considerated if I done something wrong.

sti0 committed:
Added new statistic attributes

Added new attributes:
  - Cleaning count
  - Total cleaning time
  - Total cleaning area
  - Time left to change main brush, side brush and filter

sti0 committed:
Added new statistic attributes

Added new attributes:
  - Cleaning count
  - Total cleaning time
  - Total cleaning area
  - Time left to change main brush, side brush and filter

homeassistant commented:
Hi @sti0, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9433).

Once you do that we will be able to review and accept this pull request.

Thanks!

sti0 committed:
Code corrections

Code corrections

sti0 committed:
Code corrections

Code corrections

sti0 commented:
I don't know what to do to pass Travis CI. Could someone help me?

Danielhiversen commented:
First of all,  vacuum/xiaomi.py has been renamed to vacuum/xiaomi_miio.py . So you should rebase your branch on the latest dev version, and fix the merging conflict.

You also have some indentation errors
https://travis-ci.org/home-assistant/home-assistant/jobs/275650705

sti0 commented:
Ok. Thank you @Danielhiversen for the hint. I will fix it tomorrow.

I'm sorry that I'm not so familiar with this dev tools.

sti0 committed:
Remove wronge hanging indentation

sti0 committed:
Remove wronge hanging indentation

sti0 committed:
Added new attributes

ATTR_MAIN_BRUSH_LEFT
ATTR_SIDE_BRUSH_LEFT
ATTR_FILTER_LEFT
ATTR_CLEANING_COUNT
ATTR_CLEANED_TOTAL_AREA
ATTR_CLEANING_TOTAL_TIME

sti0 committed:
Remove trailing white space

pvizeli commented on homeassistant/components/vacuum/xiaomi_miio.py:
```diff
@@ -195,7 +201,17 @@ def device_state_attributes(self):
+                    self.consumable_state.main_brush_left),
+                ATTR_SIDE_BRUSH_LEFT: str(
+                    self.consumable_state.side_brush_left),
+                ATTR_FILTER_LEFT: str(self.consumable_state.filter_left)})
```
do this value change every call?

sti0 replied:
if the vacuum is running, this values change every time. Because the more he works, the less u can use the brushes and filter.

Commit:
Corrections of the unit test for new attributes

Commit:
Hound corrections

Commit:
Init self.clean_history, self.consumable_state

Commit:
Init self.clean_history, self.consumable_state

Commit:
Init self.clean_history, self.consumable_state

Commit:
Init self.clean_history, self.consumable_state

Commit:
Hound correction

Commit:
Hound correction

Commit:
Hound correction

Commit:
Hound correction

MartinHjelmare commented on homeassistant/components/vacuum/xiaomi_miio.py:
```diff
@@ -344,8 +363,14 @@ def async_update(self):
         from mirobo import DeviceException
         try:
             state = yield from self.hass.async_add_job(self._vacuum.status)
+            consumable = yield from self.hass.async_add_job(
```
You can just set the instance attribute `self.consumable_state` directly.

sti0 replied:
Thanks. I will change that tonight.

MartinHjelmare commented on homeassistant/components/vacuum/xiaomi_miio.py:
```diff
@@ -344,8 +363,14 @@ def async_update(self):
             state = yield from self.hass.async_add_job(self._vacuum.status)
+            consumable = yield from self.hass.async_add_job(
+                self._vacuum.consumable_status)
+            cleanhis = yield from self.hass.async_add_job(
```
Same.

pvizeli commented:
Can you filter the state values to see only days?
Otherwiese you floot the state tables and database with changes for attributes

sti0 commented:
I would suggest setting the values to hours to display how long until the brush / filter Change left.

In addition I would set the total cleaning time to full minutes and the total cleaned area to rounded square meter.

The question is, if I should also change the existing values - clean time and clean area, which show the actual values - on rounded square meter and full minutes?

rytilahti commented:
Just a side note, it may be useful to provide the non-formatted versions of those values (the returned time values are `timedelta`s from which you can use `seconds` to extract the precise time.

Commit:
- Cleaning time and total cleaning time shown in minutes
- Cleaned area and total cleaned area shown in square meters
- Main brush left, side brush left, filter left time shown in hours
- Display of the unit of measurement

Commit:
Remove trailing white spaces

Commit:
Fixed wrong continued indentation

Commit:
Fixed Hound

Commit:
Fixed Hound

Commit:
Remove UOM

MartinHjelmare commented on homeassistant/components/vacuum/xiaomi_miio.py:
```diff
@@ -194,8 +203,26 @@ def device_state_attributes(self):
-                ATTR_CLEANED_AREA: round(self.vacuum_state.clean_area, 2)})
+                ATTR_CLEANING_TIME: str(int(
+                    self.vacuum_state.clean_time.total_seconds()
+                    / 60)) + ' min',
```
I'd suggest not adding units to the attributes. Values without units are easier to parse in a template for the users.

sti0 replied:
Is there any other method to declare the units? The problem that I see is that the user doesn't know which unit it is. You have minutes, hours and square meters. The biggest problem would be to distinguish hours and minutes. Or I just mean that is a problem?

MartinHjelmare replied:
At minimum we should make the units clear in the docs. I don't think there's another way to show it in the GUI. I'm not sure what's worse, and I don't think we have a rule for this yet in home assistant, but I would still favor my suggestion . 3rd person, or more, input is welcome.

sti0 replied:
I agree with you. It is easier to work with the attribute values without units. Maybe this could be a new feature to add units to attributes to show them in the gui.

sti0 commented:
something went absolutely wrong -.-

How could I remove all these commits now?

rytilahti commented:
You have to rebase your commits on top of the upstream/dev, something like:
```
git fetch upstream
git rebase upstream/dev
```
should work.

Commit:
Merge

sti0 commented:
@rytilahti I've done that but commits are always at this pull request... :-( I don't get it...

armills commented:
This generally happens when you work off of the dev branch instead of creating a topic branch. (See here: https://home-assistant.io/developers/development_submitting/ ) Then, if you have conflicts you want to follow the instructions here: https://home-assistant.io/developers/development_catching_up/

Right now the best way to recover is probably to run these commands:
```bash
$ git branch temp-branch # Create a temporary branch to recover in case something goes wrong.
$ git rebase -i upstream/dev # Delete any commits here that you didn't create.
$ git push origin -f # Since you used rebase to rewrite the commit history, you will need to force-push to your GitHub fork.
```

Commit:
Removed double declarations

sti0 commented:
@armills Thank you so much. This did work  :-)
